### ISSUE TITLE
Time doesn't display reliably on watch

### ISSUE THREAD

**Issue by andyburnett** [Knowinnovation/kitime#31]
Start, Pause, Reset buttons do work. However, the countdown display is only visible intermittently. It also runs at about double speed.

**Comment by drewsdunne** [comment]
Can you expand on the intermittently part? And hmm okay I'll have to check again I thought I fixed the twice the speed bug but I guess not.

**Comment by andyburnett** [comment]
Well, most of the time it doesn't display at all. However, if you select a
timer - on the phone - and fire up the watch, you do see the time. It also
correctly shows the time if you adjust the timer - before starting.
However, as soon as it starts, the time vanishes from the watch.
Pause/Play still works, but you don't see anything.

**Comment by drewsdunne** [comment]
Alright I was afraid of this issue. And yes I thought I mentioned when I originally made the watch app that you must select a timer on the phone to use the watch.

**Comment by andyburnett** [comment]
Needing to select an active timer makes perfect sense. I am curious about why the time vanishes. Do you think it is a problem with the label, or the communication. I.e. is the time still displayed, but it is displaying nothing, or does the label vanish for some reason

**Comment by drewsdunne** [comment]
I had been experiencing this problem before and I tried to fix it but I guess it only fixed on the simulator. I don't think it likes the constant update to the label.

**Comment by andyburnett** [comment]
If you build the watch app without the firebase elements, i.e. the simplest
version we can think of, does the problem persist? I am wondering whether
we are solving the right problem.

**Comment by drewsdunne** [comment]
Well I need firebase to have the label.

**Comment by andyburnett** [comment]
Ah, I was thinking about this a very simple test case to try to determine where the problem lay

**Comment by drewsdunne** [comment]
I read some other people have been having this problem. I think it's a problem with the label updating too frequently.

**Comment by andyburnett** [comment]
OK, so we need a new plan. I think we could:

   1. Wait for the next watch os release and see if the problem has gone
   away
   2. Try something other than label. For instance, can we draw our own
   numbers from graphics that we throw up on the screen?

Cheers

**Comment by drewsdunne** [comment]
Well there might be a wait to fix it now, I had tried updating the label asynchronously but there might be a solution online that I just have to dig for. But drawing our own numbers is a little out of the question, that would requiring recreating the whole label object.

**Comment by andyburnett** [comment]
Ok, I think we will just park this issue for the time being. Let's fix the other bugs and call it a release. 

I have had a similar problem with a previous app. The trick is to create a numeral image strip for each digit, and then use eg the minute value as the image index for that digit. 

Probably easier to code than write in English!!

Anyway let's hope we get a proper fix sometime.

**Comment by drewsdunne** [comment]
Ohh I see what you mean. That actually might not work on the watch because they don't encourage using images on the watch app. You only get so much storage because the watch apps have to be so small.
